### ISSUE TITLE
Somehow RxSwift will be treated as dynamic framework

### ISSUE THREAD

**Issue by muukii** [tuist/tuist#6740]
### What happened?

Despite of RxSwift has both types for library as dynamic and static, Tuist makes a project for RxSwift as dynamic framework.

[rxswift-dynamic 2.zip](https://github.com/user-attachments/files/17038568/rxswift-dynamic.2.zip)

![CleanShot 2024-09-18 at 11 21 00@2x](https://github.com/user-attachments/assets/5406ac5a-c9d6-421c-9634-8afcdc55e576)

### How do we reproduce it?

attached zip that reproduces this case

### Error log

no

### macOS version

14.6.1

### Tuist version

https://tuist-community.slack.com/archives/C018QG7U7SN/p1726626095777989

4.26

### Xcode version

16.0

**Comment by fortmarek** [comment]
@Arideno I thought this [PR](https://github.com/tuist/tuist/pull/6559) should have fixed this. Any ideas why `RxSwift` default is still `.framework` and not `.staticFramework`?

**Comment by alpha-moisol** [comment]
@fortmarek as I see, `RxSwift` is hardcoded to dynamic library [https://github.com/tuist/tuist/blob/3572653067b273f7eaa3823321ebd2b46f52f692/Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift#L273](here)

**Comment by Arideno** [comment]
@fortmarek as I see, RxSwift is hardcoded to dynamic library [here](https://github.com/tuist/tuist/blob/3572653067b273f7eaa3823321ebd2b46f52f692/Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift#L273)

**Comment by muukii** [comment]
wow, interesting

**Comment by fortmarek** [comment]
Ahh, I see, I don't know how I missed that 😶‍🌫️ 

@muukii can you try if `RxSwift` works as a static framework? I believe that hardcode mapping should no longer be necessary. You can override it in `PackageSettings`.

**Comment by muukii** [comment]
okay. I'm on ite

**Comment by muukii** [comment]
yes it works!

**Comment by muukii** [comment]
I think we can close this issue for now. Thank you.

**Comment by fortmarek** [comment]
@muukii would you be up for updating the hardcoded mapping? It looks it's not necessary anymore 🙂

**Comment by muukii** [comment]
sure

**Comment by ghGarnik** [comment]
Hey mates, I think I'm experiencing the same on my side after updating from 4.21.2 to the latest version. Actually, it's happening since 4.22.0 so I guess it's related to [this PR ](https://github.com/tuist/tuist/pull/6559) as you pointed out.

I'm not specifying anything in `productTypes` in the `Package.swift` to let Tuist go for the default fallback. In the left you can see how it was the embed and sign strategy set automatically, on the right almost all except for RxSwift itself are set to Do Not Embed. 

<img width="500" alt="Framework type" src="https://github.com/user-attachments/assets/15b533e2-fb40-4fa0-9ec1-c2b762e71f11">

Now I'm having crashes in the app related to linking like the following one.
Given how it was linked before, should I manually specify the product type from now on? Or will the removing of the hardcoded value you mentioned above solve it? 
If it's not related to this, please could you point out where to look for?
```
objc[99158]: Class _TtC7RxCocoa25RxScrollViewDelegateProxy is implemented in both (0x103081d60) and (0x105054388). One of the two will be used. Which one is undefined.
```

**Comment by Arideno** [comment]
Hello, @ghGarnik. As I see there is RxCocoa-Dynamic library in RxCocoa's Package.swift file. So you should use this one instead of RxCocoa if you want it to be dynamic library

**Comment by ghGarnik** [comment]
yup, the dynamic versions have been in the RxSwift Package for 3 years. I've been with Tuist dependencies since it's available, without specifying any product type for Rx, so I was wondering how the changes in `4.22.0` are resolving differently

**Comment by ghGarnik** [comment]
> Hello, @ghGarnik. As I see there is `RxCocoa-Dynamic` library in RxCocoa's Package.swift file. So you should use this one instead of `RxCocoa` if you want it to be dynamic library

Thanks for the replies @Arideno 🙏
Even using that, it's not grabbing the product type specified in the Package.swift of Rx, where it specifies ` .library(name: "RxCocoa-Dynamic", type: .dynamic, targets: ["RxCocoa"]),` unless I manually specify it in the packageSettings as `productTypes: ["RxCocoa": .framework],`

Additionally, RxCocoa also dependes on `RxCocoaRuntime` which is also static-linked now unless I explicitly set it too. 
Is there any way to opt-out this behaviour? IMO it's a breaking change, devs should be able to choose rather the automatic behaviour or the default to static.

Btw, if this not related to the OP issue, please let me know and I'll open a separate issue

**Comment by Arideno** [comment]
@ghGarnik as I understand the name of the product is different, `RxCocoa` dependency will result in static framework and `RxCocoa-Dynamic` will result in dynamic framework

**Comment by ghGarnik** [comment]
AFAICS `RxCocoa-Dynamic` it's just the library name, but the product is still `RxCocoa`. I've tried using it everywhere like `.external(name: "RxCocoa-Dynamic")` and it still links `RxCocoa` static.

Only solution for me was to explicitly set the product type like: 
```
let packageSettings = PackageSettings(
    productTypes: ["RxCocoa": .framework,
                   "RxCocoaRuntime": .framework,
                   "RxRelay": .framework,
                   "RxBlocking": .framework,
                   "RxTest": .framework])
```

 `RxCocoaRuntime` is not even referenced explicitly in my project. It's an internal dependency of RxCocoa. Before it was automatically linked dynamic. Now it's static unless I explicitly set it

**Comment by leeari95** [comment]
@fortmarek 

Hi, I've encountered the following runtime error as well:

```
objc[99158]: Class _TtC7RxCocoa25RxScrollViewDelegateProxy is implemented in both (0x103081d60) and (0x105054388). One of the two will be used. Which one is undefined.
```
From the discussion above, it seems like the current behavior of setting RxSwift as a dynamic library might be updated in a future release. Could you clarify if this update is planned?

In the meantime, as a workaround, I've manually configured RxSwift as a static library to resolve the issue. 
Thanks for your help!

**Comment by leeari95** [comment]
Hi @ghGarnik,

I encountered a similar issue with RxCocoa, and I found that adding the `-ObjC` flag to the `OTHER_LDFLAGS` in the build settings resolved the problem without needing to explicitly set the product types to `.framework`.

You can add the flag in your project settings or via Tuist like this:

```swift
let project = Project(
    name: "YourProject",
    settings: Settings(
        base: [
            "OTHER_LDFLAGS": "-ObjC"
        ]
    ),
    ...
)
```

This ensures that Objective-C symbols used internally by RxCocoa and its dependencies (like `RxCocoaRuntime`) are properly linked at runtime.

Hope this helps! Let me know if it works for your case. 😊

**Comment by dilshodzopirov** [comment]
> AFAICS `RxCocoa-Dynamic` it's just the library name, but the product is still `RxCocoa`. I've tried using it everywhere like `.external(name: "RxCocoa-Dynamic")` and it still links `RxCocoa` static.
> 
> Only solution for me was to explicitly set the product type like:
> 
> ```
> let packageSettings = PackageSettings(
>     productTypes: ["RxCocoa": .framework,
>                    "RxCocoaRuntime": .framework,
>                    "RxRelay": .framework,
>                    "RxBlocking": .framework,
>                    "RxTest": .framework])
> ```
> 
> `RxCocoaRuntime` is not even referenced explicitly in my project. It's an internal dependency of RxCocoa. Before it was automatically linked dynamic. Now it's static unless I explicitly set it

I can't thank you enough! You saved my week ))

**Comment by dilshodzopirov** [comment]
> [@dilshodzopirov](https://github.com/dilshodzopirov) so did you experience the same? Were you in a previous version of Tuist and it started happening after update? Could you please provide a bit of info to understand when is this happening?

I wasn't using Tuist before, just SPM. After I setup Tuist it stopped working until I put RxSwift as framework in packageSettings
